react/LocationSearch.tsx

```diff
@@ -127,11 +137,16 @@ const LocationSearch: React.FC<LocationSearchProps> = ({
   }
 
   const handleAddressSelection = (selectedAddress: string) => {
-    setSearchTerm(selectedAddress)
     const id = suggestions.find(
       address => address.description === selectedAddress
     )?.externalId
-    executeGetAddressByExternalId({ variables: { id } })
+
+    if (id) {
+      setSearchTerm(selectedAddress)
+      executeGetAddressByExternalId({ variables: { id } })
+    } else {
+      console.error(`${selectedAddress} was not found`)
+    }
```

Comment:
I did this to avoid a possible race condition... but is it even possible?

Reply:
Well, I don't know the methods so deeply to properly discuss about it, but if by "this" you mean the if/else statement, I don't see how this could avoid a race condition. The way I see it is simply saving up one possibly unnecessary request (if there wasn't an id). It's useful though.